packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap

```diff
@@ -6,6 +6,7 @@ exports[`form-render utils should generate before & complete types if datastore
     onSuccess?: (fields: Record<string, unknown>) => void;
     onError?: (fields: Record<string, unknown>, errorMessage: string) => void;
     onCancel?: () => void;
+    onChange?: (fields: Record<string, unknown>) => Record<string, unknown>;
```

Comment:
Shouldn't the type be specifically related to the fields in the form? We should be able to infer all of those.

Reply:
I can create a follow up pr to create the types for datastore-backed forms, though for custom forms it's bit more unknown as we'd need a larger task to track those data types.
cc: @hein-j

Reply:
Even for JSON forms though, we still know the types that the different primitives return so we can still provide a map of fields to those types.